src/Mod/Arch/ArchReference.py

```diff
@@ -183,14 +183,14 @@ def cleanShape(self,shapedata,obj,materials):
             for edge in shape.Edges:
                 found = False
                 for solid in shape.Solids:
-                    if found:
-                        break
                     for soledge in solid.Edges:
-                        if found:
-                            break
                         if edge.hashCode() == soledge.hashCode():
                             found = True
                             break
+                    if found:
+                        break
+                if found:
+                    break
```

Comment:
@yorikvanhavre can you double-check this change? It's not just a reformat, it is an algorithm change: the outermost for loop has an `else` clause, which only makes sense if there is a `break` statement that affects that outer loop. So I guessed at what that clause might be, but I'm not 100% sure of this one.

Reply:
Yes, you are right... Indeed my implementation was not good, the else clause would almost never be triggered when it needs. I think yours fixes the problem